jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglWindow.java

```diff
@@ -271,12 +324,91 @@ public void callback(long args) {
             glfwSwapInterval(0);
         }
 
+        final Object[] icons = settings.getIcons();
+
+        if (icons != null) {
+
+            final GLFWImage[] images = imagesToGLFWImages(icons);
+            final GLFWImage.Buffer iconSet = GLFWImage.malloc(images.length);
+
+            for (int i = images.length - 1; i >= 0; i--) {
+                final GLFWImage image = images[i];
+                iconSet.put(i, image);
+            }
+
+            glfwSetWindowIcon(window, iconSet);
+
+            iconSet.free();
+
+            for (final GLFWImage image : images) {
+                image.free();
+            }
+        }
 
         glfwShowWindow(window);
 
         allowSwapBuffers = settings.isSwapBuffers();
+    }
 
-        // TODO: When GLFW 3.2 is released and included in LWJGL 3.x then we should hopefully be able to set the window icon.
+    /**
+     * Convert array of images to array of {@link GLFWImage}.
+     */
+    private GLFWImage[] imagesToGLFWImages(final Object[] images) {
+
+        final GLFWImage[] out = new GLFWImage[images.length];
+
+        for (int i = 0; i < images.length; i++) {
+            final BufferedImage image = (BufferedImage) images[i];
+            out[i] = imageToGLFWImage(image);
+        }
+
+        return out;
+    }
+
+    /**
+     * Convert the {@link BufferedImage} to the {@link GLFWImage}.
+     */
+    private GLFWImage imageToGLFWImage(BufferedImage image) {
+
+        if (image.getType() != BufferedImage.TYPE_INT_ARGB_PRE) {
+
+            final BufferedImage convertedImage = new BufferedImage(image.getWidth(), image.getHeight(), BufferedImage.TYPE_INT_ARGB_PRE);
+            final Graphics2D graphics = convertedImage.createGraphics();
+
+            final int targetWidth = image.getWidth();
+            final int targetHeight = image.getHeight();
+
+            graphics.drawImage(image, 0, 0, targetWidth, targetHeight, null);
+            graphics.dispose();
+
+            image = convertedImage;
+        }
+
+        final IntBuffer w = BufferUtils.createIntBuffer(1);
+        final IntBuffer h = BufferUtils.createIntBuffer(1);
+        final IntBuffer comp = BufferUtils.createIntBuffer(1);
+
+        final ByteArrayOutputStream bout = new ByteArrayOutputStream(4096);
+        try {
+            ImageIO.write(image, "png", bout);
```

Comment:
Get rid of `STBImage.stbi_load_from_memory`, you don't need it if the data is loaded directly from the `BufferedImage`